pallene/checker.lua

```diff
@@ -687,7 +689,7 @@ function Checker:check_exp_synthesize(exp)
             check_type_is_condition(exp.exp, "'not' operator")
             exp._type = types.T.Boolean()
         else
-            error("impossible")
+            typedecl.tag_error(tag, string.format("unknown unary operator '%s'.", op))
```

Comment:
With the new default error message I proposed, I think we could use `typedecl.tag_error(op)`.